src/main/java/eggventory/logic/parsers/ParseAdd.java

```diff
@@ -151,9 +151,13 @@ public Command processAddTemplate(String input) throws BadInputException {
      * @param input string in the format matricNo, stockCode and quantity.
      * @return a command to add a loan.
      */
-    private Command processAddLoan(String input) {
-        String[] addInput = input.split(" +");
+    private Command processAddLoan(String input) throws InsufficientInfoException {
+        if (!Parser.isCommandComplete(input, 2)) {
```

Comment:
read javadoc for isCommandComplete :(

Reply:
Thanks, there was a logical error. Anyways when you see the new code, it basically works as such:

1)  AddLoanByTemplate takes in 2 arguments (Matric no, TemplateName)
2)  AddLoan takes in 3 arguments (Matric No, StockCode, Quantity)

Hence, it is essential to add a check after step 1, whether the input String in processAddLoan has 3 arguments, otherwise throw a InsufficientInfoException. Otherwise when it reaches the line

 "return new AddLoanCommand(CommandType.ADD, addInput[0], addInput[1], Integer.parseInt(addInput[2]));" 

it will return a IndexOutOfBound error because addInput[2] is out of bound. Which was why the JUnit test failed.